### PR TITLE
[SHEPHERD] Interprocess LCM api implemented

### PR DESCRIPTION
**Pull request by charleshong3**
Implemented LCM API for Shepherd

### CODE REVIEW

File: shepherd/LCM.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import lcm
+import threading
+import json
+
+class LCMClass:
```
Comment (JHGZhou) [inline shepherd/LCM.py]: Add a comment here describing how this class works very generally. Also describe how to make use of the LCM object.

File: shepherd/LCM.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        def __init__(self, queue, receive_channel):
+            self.queue = queue
+            self.receive_channel = receive_channel
+            self.lc = lcm.LCM()
+
+        # def start_thread(self):
```
Comment (JHGZhou) [inline shepherd/LCM.py]: remove unneeded code

File: shepherd/LCM.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            rec_thread = threading.Thread()
+            rec_thread.run = run
+            rec_thread.start()
+
+        # Send a list into a target
+        def lcm_send(self, target_channel, header, *args):
+            msg = '|'.join(str(a) for a in args)
+            msg = '|'.join([header, msg])
+            self.lc.publish(target_channel, msg.encode())
+        """
```
Comment (JHGZhou) [inline shepherd/LCM.py]: remove uneeded code

File: shepherd/LCM.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+            def run():
+                while True:
+                    self.lc.handle()
+
+            rec_thread = threading.Thread()
+            rec_thread.run = run
+            rec_thread.start()
+
+        # Send a list into a target
```
Comment (JHGZhou) [inline shepherd/LCM.py]: comment style should be three single quotes above and below the docstring and the docstring should be under the def statement and above the rest of the method

**Review by JHGZhou — CHANGES_REQUESTED** [review]

**Review by brandonxxlee — DISMISSED** [review]
lgtm, Just make sure you test thoroughly.

**Comment by kevinmasd** [comment]
Handled in #473

**Pull request closed without merging**